lighthouse-core/test/lib/page-functions-test.js

```diff
@@ -59,4 +59,31 @@ describe('Page Functions', () => {
       assert.equal(pageFunctions.getNodeSelector(childEl), 'div#wrapper > div.child');
     });
   });
+
+  describe('getNodeLabel', () => {
+    it('Returns innerText if element has visible text', () => {
+      const el = dom.createElement('div');
+      el.innerText = 'Hello';
+      assert.equal(pageFunctions.getNodeLabel(el), 'Hello');
+    });
+    it('Falls back to children and alt/aria-label if a title can\'t be determined', () => {
+      const el = dom.createElement('div');
+      const childEl = dom.createElement('div', '', {'aria-label': 'Something'});
+      el.appendChild(childEl);
+      assert.equal(pageFunctions.getNodeLabel(el), 'Something');
+    });
+    it('Truncates long text', () => {
+      const el = dom.createElement('div');
+      el.setAttribute('alt', Array(100).fill('a').join(''));
+      assert.equal(pageFunctions.getNodeLabel(el).length, 80);
+    });
+    it('Uses tag name for html tags', () => {
+      const el = dom.createElement('html');
+      assert.equal(pageFunctions.getNodeLabel(el), 'html');
+    });
+    it('Uses tag name if there is no better label', () => {
+      const el = dom.createElement('div');
+      assert.equal(pageFunctions.getNodeLabel(el), 'div');
+    });
+  });
```

Comment:
also add a test for the recursive case?

Reply:
We already have the `Falls back to children` test case above, do you mean checking it falls back to the parent tag name if neither the parent nor any children are useful? I've modified the test case to check that.

Reply:
> We already have the `Falls back to children` test case above, do you mean checking it falls back to the parent tag name if neither the parent nor any children are useful? I've modified the test case to check that.

whoops, no, that's me not reading carefully enough. I like the new change too, though!